### DISCUSSION TITLE
How to use 2 or more custom Fact attributes

### DISCUSSION THREAD

**Discussion by dluc** [xunit/xunit#2257] [Question] [answered]
Is there some guidelines or examples about how to use 2 custom Fact attributes? 

For instance, the following code doesn't compile, complaining that "Test methods cannot have multiple Fact or Theory attributes"

```csharp
[SkippableFact] [RetryFact]
public void SomeTest()
{
    // some code
}
```

**Comment by bradwilson** [comment, marked as answer]
You cannot mix these like this; each Fact-derived attribute has a discoverer attached to it that it is also responsible for understanding how to run the test. Having two of these would, at best, run the test twice, once with each behavior.

You need a new custom Fact-derived attribute (and discoverer, and runner) which exhibits the (combine) behavior you want.
